dotnet-client-libraries/ITextBasics/ITextBasics/BigDocument.cs

```diff
@@ -0,0 +1,87 @@
+﻿using iText.IO.Image;
+using iText.Kernel.Pdf;
+using iText.Layout;
+using iText.Layout.Element;
+using iText.Layout.Properties;
+using System.Diagnostics;
+using System.Reflection;
+
+namespace ITextBasics
+{
+    public class BigDocument
+    {
+        public void Create(string pdfFileName)
+        {
+            using var writer = new PdfWriter(pdfFileName);
+            using var pdfDocument = new PdfDocument(writer);
+            using var document = new Document(pdfDocument);
+
+            string? executablePath = Path.GetDirectoryName(Assembly.GetEntryAssembly()!.Location);
+            string imageFile = Path.Combine(executablePath!, "Resources", "Flower.jpg");
+
+            for (int i = 0; i < 3; i++)
+            {
+                document.Add(CreateHeader1(LoremIpsum(3)));
+                document.Add(CreateHeader2(LoremIpsum(7)));
+                document.Add(CreateNormalParagraph(LoremIpsum(100)));
+
+                document.Add(CreateHeader2(LoremIpsum(7)));
+                document.Add(CreateNormalParagraph(LoremIpsum(55)));
+                document.Add(CreateNormalParagraph(LoremIpsum(27)));
+
+                document.Add(CreateHeader1(LoremIpsum(6)));
+                document.Add(CreateHeader2(LoremIpsum(3)));
+                document.Add(CreateNormalParagraph(LoremIpsum(50)));
+                document.Add(CreatePictureParagraph(imageFile));
+                document.Add(CreateHeader2(LoremIpsum(3)));
+                document.Add(CreateNormalParagraph(LoremIpsum(45)));
+            }
+        }
+
+        private Paragraph CreateHeader1(string caption)
+        {
+            return new Paragraph(caption)
+                .SetTextAlignment(TextAlignment.CENTER)
+                .SetFontSize(20)
+                .SetBold();
+        }
+
+        private Paragraph CreateHeader2(string caption)
+        {
+            return new Paragraph(caption)
+                .SetTextAlignment(TextAlignment.LEFT)
+                .SetFontSize(15)
+                .SetBold();
+        }
+
+        private Paragraph CreateNormalParagraph(string text)
+        {
+            return new Paragraph(text);
+        }
+
+        private Paragraph CreatePictureParagraph(string imageFile)
+        {
+            var imageData = ImageDataFactory.Create(imageFile);
+            var image = new Image(imageData);
+
+            return new Paragraph()
+                .Add(image);
+        }
+
+        private string LoremIpsum(int numberOfWords)

```

Comment:
You could make this method static.

Reply:
Yes, that is true. I will make it static.

Any function that does not use any internal object state can be static. But should it be? If this would be "my program" I would not make it static just because otherwise I would have to go through the codebase and make any function that can be static into a static function.

Reply:
As a best practice, methods which do not access instance data should be static. Both VS and ReSharper will point this out as a violation of code analysis rule [CA1822](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822).

Reply:
I repaired all methods that can be static to static.